### DISCUSSION TITLE
Cruiser 3.0.23 - force close

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#216] [Q&A] [answered]
I just checked on two devices: Allow the screen to go blank while BRouter is counting alternative routes (about 70 km). After a few minutes, turn on the screen; one route is drawn correctly, the second alternative, is as a straight line. Just touch the straight line and the app closes.

**Comment by devemux86** [comment]
@Ciutnik thanks for the report!

(I still do not see anything in Google Play Console)

If you don't close the screen and watch the route calculation until the finish, what happens?

Can you share the route, so I can test it?

**Reply by Ciutnik** [reply]
As long as I don't let the screen turn off by itself, everything is fine. 

Since there must be a screen auto-blank before BRouter finishes counting, either there must be four alternatives or a long route, or a short screen timeout. If any of these conditions are met, it happens on every route. Maybe it's the fault of the Samsung tablets on which I'm testing the application?

Edit: This happened while the screen was active but Cruiser was running in the background!

<img src="https://github.com/devemux86/cruiser/assets/122647782/516fadd7-2b1f-4806-90f0-58306042f60c" width="50%"/>

**Comment by devemux86** [comment]
I start a BRouter calculation of a long route with 4 alternatives, close the device screen or put Cruiser in background, but it works.
It does not return any straight lines.

@Ciutnik can you share a specific route example and the BRouter settings?

Info: BRouter (offline) may timeout with multiple avoidances if it cannot find available roads.
So probably some of your alternatives failed.

---

However, there is need for a fix and it will be included in the next Cruiser:
Only the successful routes should appear in the results, not the failed ones (straight lines).
(this should also fix the crash)

**Reply by Ciutnik** [reply]
I just caused the app to close with this route and these app settings. As I wrote, if I keep Cruiser in the foreground, and the screen stays on, everything is OK, so maybe some aggressive process killing is working here. However, I have turned off all "resource savers" (but maybe it just seems that way to me...)

Thanks for addressing the issue, as you can see specific.

[Cruiser_straight_line_error.zip](https://github.com/devemux86/cruiser/files/11749108/Cruiser_straight_line_error.zip)

**Comment by devemux86** [comment, marked as answer]
The fix for BRouter alternative routes is in [Cruiser 3.0.24](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
I have not managed to reproduce a failed background calculation of BRouter alternative route.
Could be calculation timeout, background memory restriction, ...?

However we changed the BRouter driving profile in Cruiser, so we better check it again:
- https://github.com/devemux86/cruiser/discussions/175#discussioncomment-6174360

**Comment by Ciutnik** [comment]
Problem cleverly solved! I have not been able to draw a troublesome straight line. Only instead of e.g. four alternatives there are three or two. But if that's the price of stability, it's absolutely worth it! 😀 Thanks a lot! 👍

**Comment by Ciutnik** [comment]
Force close again today. 
Because I wanted to see more of the map already at the speed of ~80 km/h, I decided to change the max speed parameter to 80 km/h. In the settings search (**brilliant feature!**) I found the right place, and after saving the change, the application closed unexpectedly. This time I was able to send a report, which I did (at 11:50 CET). Since I was on tour I couldn't add any description so I'm doing it now.

Other than that, the Cruiser performs really well. Today I did my first trip 100% offline using only BRouter. The convenience and reliability of such a solution compensates for the lack of street names that I miss. Previously, I planned trips with Kurviger and imported a file, but this solution did not suit me very much... I'm not likely to go back to that workflow, so I guess I can say "The king is dead, long live the king!"

**Reply by devemux86** [reply]
@Ciutnik thanks for the report!

(currently I do not see any new report in Google Play)

Do you mean you changed the max speed setting in your selected profile,
because you wanted a different scale in the navigation auto zoom?

Can you describe the exact steps and what the app was doing at that time?
Perhaps send me the app settings .pref file or the route if needed?

Also did you have many favorites visible?

**Reply by devemux86** [reply]
If you had the alternatives enabled in BRouter, please send me the route and the app .pref file.

Maybe we need more improvements in the BRouter results.
Did you do a rerouting or it was simply a change in settings?

**Reply by Ciutnik** [reply]
> Do you mean you changed the max speed setting in your selected profile,
because you wanted a different scale in the navigation auto zoom?

Yes, I just did that (it happened here: 51.949430112763125,20.641797963956403), for a moment the screen turned white, and then app shut down. The Android error report sending window appeared, which I had to confirm and agree to send the data. Too bad you didn't get anything. But maybe it was a single and unrepeatable error. 

I didn't change the route or anything else. I only changed 100 km/h to 80 km/h. I found the right setting with the "magnifying glass" tool. There are literally a dozen of favorites displayed. Most are hidden.

<img src="https://github.com/devemux86/cruiser/assets/122647782/3b21c3d1-7023-4783-85c5-36d6eaa68a31" width="50%"/>

I include pref and route:

[Cruiser_pref_and_route.zip](https://github.com/devemux86/cruiser/files/11992402/Cruiser_pref_and_route.zip)

**Reply by Ciutnik** [reply]
You know, I'm testing different options and settings very intensively right now. It's possible that I used some killer mix 🙂 It doesn't change the fact that in my eyes the Cruiser is like a rock. Really kudos for your work👍

**Reply by devemux86** [reply]
@Ciutnik thanks for the files! I loaded and checked them, all seem ok.

I still do not see anything in Google Play.
Were you online when sent the report?

> You know, I'm testing different options and settings very intensively right now. It's possible that I used some killer mix :slightly_smiling_face: 

Everyone, please push Cruiser to its limits!
So we can find any possible issue and improve it.

> It doesn't change the fact that in my eyes the Cruiser is like a rock. Really kudos for your work :+1: 

Many thanks for your support! :slightly_smiling_face:

**Reply by devemux86** [reply]
> I only changed 100 km/h to 80 km/h. I found the right setting with the "magnifying glass" tool.

@Ciutnik thanks, after more tests, this description helped me to reproduce your report! :+1: 

Indeed we need a fix here, it will be available in the next Cruiser update.

**Reply by Ciutnik** [reply]
I'm glad I could help you locate the source of the error and make Cruiser even more reliable! 👍

**Reply by devemux86** [reply]
It should be fixed in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
